### PR TITLE
Implemented exception chaining for all re-raised exceptions

### PR DESCRIPTION
**Pull request by alexthomas93**
# Description

This is a small PR which adds the `from` keyword to all re-raised exceptions. The effect of this is to make the resulting stack traces more readable i.e. traces such as this
```bash
⋮
raise Neo4jError.hydrate(**metadata)
neo4j.exceptions.ClientError: {code: Neo.ClientError.Schema.EquivalentSchemaRuleAlreadyExists} {message: An equivalent index already exists, 'Index( id=1, name='movie_tagline_embeddings', type='VECTOR', schema=(:Movie {taglineEmbedding}), indexProvider='vector-2.0' )'.}

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/alexthomas/Documents/neo4j-genai-python/a.py", line 20, in <module>
    create_vector_index(
  File "/Users/alexthomas/Documents/neo4j-genai-python/src/neo4j_genai/indexes.py", line 83, in create_vector_index
    raise Neo4jIndexError("Neo4j vector index creation failed.")
neo4j_genai.exceptions.Neo4jIndexError: Neo4j vector index creation failed.
```
are replaced by traces such as this
```bash
⋮
raise Neo4jError.hydrate(**metadata)
neo4j.exceptions.ClientError: {code: Neo.ClientError.Schema.EquivalentSchemaRuleAlreadyExists} {message: An equivalent index already exists, 'Index( id=1, name='movie_tagline_embeddings', type='VECTOR', schema=(:Movie {taglineEmbedding}), indexProvider='vector-2.0' )'.}

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/alexthomas/Documents/neo4j-genai-python/a.py", line 20, in <module>
    create_vector_index(
  File "/Users/alexthomas/Documents/neo4j-genai-python/src/neo4j_genai/indexes.py", line 83, in create_vector_index
    raise Neo4jIndexError("Neo4j vector index creation failed.") from e
neo4j_genai.exceptions.Neo4jIndexError: Neo4j vector index creation failed.
```

## Type of Change
- [ ] New feature
- [X] Bug fix
- [ ] Breaking change
- [ ] Documentation update
- [ ] Project configuration change

## Complexity

Complexity: Low

## How Has This Been Tested?
- [X] Unit tests
- [X] E2E tests
- [X] Manual tests

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [ ] Documentation has been updated
- [ ] Unit tests have been updated
- [ ] E2E tests have been updated
- [ ] Examples have been updated
- [ ] New files have copyright header
- [X] CLA (https://neo4j.com/developer/cla/) has been signed
- [ ] CHANGELOG.md updated if appropriate

### CODE REVIEW

File: src/neo4j_genai/indexes.py [hunk 1]
```diff
@@ -65,7 +65,9 @@ def create_vector_index(
             similarity_fn=similarity_fn,
         )
     except ValidationError as e:
-        raise Neo4jIndexError(f"Error for inputs to create_vector_index {str(e)}")
+        raise Neo4jIndexError(
+            f"Error for inputs to create_vector_index {str(e)}"
```
Comment (willtai) [inline src/neo4j_genai/indexes.py:69]: would having `str(e)` in the f-string make this too verbose?

File: src/neo4j_genai/indexes.py [hunk 2]
```diff
@@ -119,7 +123,9 @@ def create_fulltext_index(
         logger.info(f"Creating fulltext index named '{name}'")
         driver.execute_query(query, {"name": name})
     except neo4j.exceptions.ClientError as e:
-        raise Neo4jIndexError(f"Neo4j fulltext index creation failed {e}")
+        raise Neo4jIndexError(
+            f"Neo4j fulltext index creation failed {e.message}"
```
Comment (willtai) [inline src/neo4j_genai/indexes.py:127]: How do you determine whether to print e.message vs e.errors()?

Comment (alexthomas93) [reply src/neo4j_genai/indexes.py:127]: `e.message` for Neo4j errors, `e.errors()` for validation errors

**Review by willtai — APPROVED** [review]
✅

**Pull request merged**